nemo/collections/tts/g2p/models/ja_jp_g2p.py

```diff
@@ -0,0 +1,157 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+from collections import defaultdict
+from typing import Dict, List, Optional, Union
+
+from nemo.collections.common.tokenizers.text_to_speech.ipa_lexicon import (
+    get_grapheme_character_set,
+    get_ipa_punctuation_list,
+)
+from nemo.collections.tts.g2p.models.base import BaseG2p
+from nemo.collections.tts.g2p.utils import set_grapheme_case
+from nemo.utils import logging
+
+
+class JapaneseG2p(BaseG2p):
+    def __init__(
+        self,
+        phoneme_dict: Union[str, pathlib.Path, Dict[str, List[str]]],
+        phoneme_prefix: str = "#",
+        ascii_letter_prefix: str = "",
+        ascii_letter_case: str = "upper",
+        word_tokenize_func=None,
+        apply_to_oov_word=None,
+        mapping_file: Optional[str] = None,
+        word_segmenter: Optional[str] = None,
+    ):
+        """
+        Japanese G2P module. This module first segments Japanese characters into words using Janome, then
+            these separated words are converted into phoneme sequences by looking them up in the 'phoneme_dict'.
+        Args:
+            phoneme_dict (str, Path, Dict): Path to jp_dict_nv_24.06.txt dict file or a dict object.
+            phoneme_prefix (str): Prepend a special symbol to any phonemes in order to distinguish phonemes from
+                graphemes because there may be overlaps between the two sets. It is suggested to choose a prefix that
+                is not used or preserved somewhere else. Default to "#".
+            ascii_letter_prefix (str): Prepend a special symbol to any ASCII letters. Default to "".
+            ascii_letter_case (str): Specify the case chosen from `"lower"`, `"upper"`, or `"mixed"`, and process the
+                cases of non-Chinese words. Default to `"upper"`.
+            word_tokenize_func: Function for tokenizing text to words.
+                It has to return List[Tuple[Union[str, List[str]], bool]] where every tuple denotes word representation
+                    and flag whether to leave unchanged or not.
+                It is expected that unchangeable word representation will be represented as List[str], other cases are
+                    represented as str.
+                It is useful to mark word as unchangeable which is already in phoneme representation.
+            apply_to_oov_word: Function that will be applied to out of phoneme_dict word.
+            word_segmenter: method that will be applied to segment utterances into words for better polyphone disambiguation.
+        """
+        assert phoneme_dict is not None, "Please set the phoneme_dict path."
+        assert (
+            word_segmenter == 'janome'
```

Comment:
any reason why choosing `janome` over `mecab`? you also need to add either in `requirements/requirements_tts.txt`, otherwise, unit tests may fail.

Reply:
I was not aware of mecab.